dali/kernels/imgproc/pointwise/linear_transformation_gpu_test.cu

```diff
@@ -0,0 +1,244 @@
+// using TestTypes = std::tuple<uint8_t, int8_t, uint16_t, int16_t, int32_t, float>;
+
+INPUT_OUTPUT_TYPED_TEST_SUITE(LinearTransformationGpuTest, TestTypes);
+
+namespace {
+
+template <class GtestTypeParam>
+using TheKernel = LinearTransformationGpu<typename GtestTypeParam::Out, typename GtestTypeParam::In,
+        kNChannelsOut, kNChannelsIn, kNDims - 1>;
+
+}  // namespace
+
+
+TYPED_TEST(LinearTransformationGpuTest, check_kernel) {
+  check_kernel<TheKernel<TypeParam>>();
+}
+
+
+TYPED_TEST(LinearTransformationGpuTest, setup_test) {
+  TheKernel<TypeParam> kernel;
+  KernelContext ctx;
+  InListGPU<typename TypeParam::In, kNDims> in(this->input_device_, this->in_shapes_);
+  auto reqs = kernel.Setup(ctx, in, make_cspan(this->vmat_), make_cspan(this->vvec_));
+  ASSERT_EQ(this->out_shapes_.size(), static_cast<size_t>(reqs.output_shapes[0].num_samples()))
+                        << "Kernel::Setup provides incorrect shape";
+  for (size_t i = 0; i < this->out_shapes_.size(); i++) {
+    EXPECT_EQ(this->out_shapes_[i], reqs.output_shapes[0][i])
+                  << "Kernel::Setup provides incorrect shape";
+  }
+}
+
+
+TYPED_TEST(LinearTransformationGpuTest, run_test) {
+  TheKernel<TypeParam> kernel;
+  KernelContext c;
+  InListGPU<typename TypeParam::In, kNDims> in(this->input_device_, this->in_shapes_);
+
+  auto reqs = kernel.Setup(c, in, make_cspan(this->vmat_), make_cspan(this->vvec_));
+
+  ScratchpadAllocator sa;
+  sa.Reserve(reqs.scratch_sizes);
+  auto scratchpad = sa.GetScratchpad();
+  c.scratchpad = &scratchpad;
+
+  OutListGPU<typename TypeParam::Out, kNDims> out(
+          this->output_, reqs.output_shapes[0].template to_static<kNDims>());
+
+  kernel.Run(c, out, in, make_cspan(this->vmat_), make_cspan(this->vvec_));
+  cudaDeviceSynchronize();
+
+  auto res = copy<AllocType::Host>(out[0]);
+  ASSERT_EQ(static_cast<int>(this->ref_output_.size()), res.first.num_elements());
+  for (size_t i = 0; i < this->ref_output_.size(); i++) {
+    EXPECT_FLOAT_EQ(this->ref_output_[i], res.second.get()[i]) << "Failed for index " << i;
+  }
+}
+
+
+TYPED_TEST(LinearTransformationGpuTest, run_test_with_roi) {
+  TheKernel<TypeParam> kernel;
+  KernelContext c;
+  InListGPU<typename TypeParam::In, kNDims> in(this->input_device_, this->in_shapes_);
+
+  auto reqs = kernel.Setup(c, in,
+                           make_cspan(this->vmat_), make_cspan(this->vvec_),
+                           make_cspan(this->rois_));
+
+  ScratchpadAllocator sa;
+  sa.Reserve(reqs.scratch_sizes);
+  auto scratchpad = sa.GetScratchpad();
+  c.scratchpad = &scratchpad;
+
+  OutListGPU<typename TypeParam::Out, kNDims> out(
+          this->output_, reqs.output_shapes[0].template to_static<kNDims>());
+
+  kernel.Run(c, out, in, make_cspan(this->vmat_), make_cspan(this->vvec_), make_cspan(this->rois_));
+  cudaDeviceSynchronize();
+
+  auto res = copy<AllocType::Host>(out[0]);
+  auto mat = color_manipulation::test::to_mat<kNChannelsOut>(this->ref_output_.data(),
```

Comment:
Why do you go through OpenCV mat here?

Reply:
I use OpenCV mat to handle ROI case, as a reference calculation method.